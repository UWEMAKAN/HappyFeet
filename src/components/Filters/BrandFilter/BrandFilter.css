.FilterBrand {
  padding: .5rem 0;
  font-size: 1.6rem;
  color: #232323;
  display: flex;
  align-content: center;
}

.FilterBrand span {
  margin-left: .5rem;
  font-size: 1.4rem;
}

.BrandFilter input,
.BrandFilter input:focus {
  outline: none;
  border: 1px solid rgba(0,0,0,.4);
  padding: .3rem;
}

@media screen and (max-width: 900px) {
  .BrandFilter input,
  .BrandFilter input:focus {
    outline: none;
    border: 1px solid rgba(0,0,0,.4);
    padding: .2rem;
    width: 15rem;
  }
}
