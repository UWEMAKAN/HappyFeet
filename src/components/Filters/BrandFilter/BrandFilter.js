import React from 'react';
import propTypes from 'prop-types';
import { FiSearch } from 'react-icons/fi';
import './BrandFilter.css';

const BrandFilter = ({ searchHandler }) => (
  <div className="Filter BrandFilter">
    <h3>Brand</h3>
    <div>
      <div className="FilterBrand">
        <FiSearch />
        <span>search brand</span>
      </div>
      <input onChange={searchHandler} name="search" id="search" type="search" />
    </div>
  </div>
);

BrandFilter.propTypes = {
  searchHandler: propTypes.func.isRequired
};

export default BrandFilter;
