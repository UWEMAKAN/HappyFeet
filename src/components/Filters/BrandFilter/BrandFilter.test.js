import React from 'react';
import { shallow } from 'enzyme';
import BrandFilter from './BrandFilter';

const renderBrandFilter = (args) => {
  const props = { ...args };
  return shallow(<BrandFilter searchHandler={props.searchHandler} />);
};

describe('BrandFilter Component', () => {
  test('should render BrandFilter', () => {
    const searchHandler = jest.fn();
    const wrapper = renderBrandFilter({ searchHandler });
    wrapper.find('input').simulate('change');
    expect.assertions(4);
    expect(wrapper.find('h3').text()).toEqual('Brand');
    expect(wrapper.find('span').text()).toEqual('search brand');
    expect(searchHandler).toHaveBeenCalledTimes(1);
    expect(wrapper).toMatchSnapshot();
  });
});
