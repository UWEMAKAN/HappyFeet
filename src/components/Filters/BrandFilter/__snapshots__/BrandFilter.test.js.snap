// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BrandFilter Component should render BrandFilter 1`] = `ShallowWrapper {}`;
